crates/biome_html_formatter/src/utils/children.rs

```diff
@@ -0,0 +1,422 @@
+        f.write_element(FormatElement::LocatedTokenText {
+            source_position: self.source_position,
+            slice: self.text.clone(),
+        })
+    }
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub(crate) enum HtmlChild {
+    /// A Single word in a HTML text. For example, the words for `a b\nc` are `[a, b, c]`
+    Word(HtmlWord),
+
+    /// A ` ` whitespace
+    ///
+    /// ```html
+    /// <div> </div>
+    /// <div>a </div>
+    /// <div> a</div>
+    /// <div>a
+    ///  b</div>
+    /// ```
+    ///
+    /// Whitespace between two words is not represented as whitespace
+    /// ```javascript
+    /// <div>a b</div>
+    /// ```
+    /// The space between `a` and `b` is not considered a whitespace.
+    Whitespace,
+
+    /// A new line at the start or end of a [HtmlText] with meaningful content. (that isn't all whitespace
+    /// and contains a new line).
+    ///
+    /// ```html
+    /// <div>
+    ///     a
+    /// </div>
+    /// ```
+    Newline,
+
+    /// A [HtmlText] that only consists of whitespace and has at least two line breaks;
+    ///
+    /// ```html
+    /// <div>
+    ///
+    ///   <test />
+    /// </div>
+    /// ```
+    ///
+    /// The text between `<div>` and `<test />` is an empty line text.
+    EmptyLine,
+
+    /// Any other content that isn't a text. Should be formatted as is.
+    NonText(AnyHtmlElement),
+}
+
+impl HtmlChild {
+    pub(crate) const fn is_any_line(&self) -> bool {
+        matches!(self, HtmlChild::EmptyLine | HtmlChild::Newline)
+    }
+}
+
+/// Creates either a space using an expression child and a string literal,
+/// or a regular space, depending on whether the group breaks or not.
+///
+/// ```html
+///  <div> Winter Light </div>;
+///
+///  <div>
+///    Winter Light
+///    Through A Glass Darkly
+///    The Silence
+///    Seventh Seal
+///    Wild Strawberries
+///  </div>
+/// ```
+#[derive(Default)]
+pub(crate) struct HtmlSpace;
+
+impl Format<HtmlFormatContext> for HtmlSpace {
+    fn fmt(&self, formatter: &mut HtmlFormatter) -> FormatResult<()> {
+        write![
+            formatter,
+            [
+                if_group_breaks(&format_args![HtmlRawSpace, soft_line_break()]),
+                if_group_fits_on_line(&space())
+            ]
+        ]
+    }
+}
+
+pub(crate) struct HtmlRawSpace;
+
+impl Format<HtmlFormatContext> for HtmlRawSpace {
+    fn fmt(&self, f: &mut Formatter<HtmlFormatContext>) -> FormatResult<()> {
+        write!(f, [text(" ")])
+    }
+}
+
+pub(crate) fn html_split_children<I>(children: I) -> SyntaxResult<Vec<HtmlChild>>
+where
+    I: IntoIterator<Item = AnyHtmlElement>,
+{
+    let mut builder = HtmlSplitChildrenBuilder::new();
+
+    for child in children {
+        match child {
+            AnyHtmlElement::HtmlContent(text) => {
+                // Split the text into words
+                // Keep track if there's any leading/trailing empty line, new line or whitespace
+
```

Comment:
Is this space intentional?

Reply:
This is mostly copied strait from the formatter for `JsxChildList`.